tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/LiferayGradleModuleProjectDeleteParticipant.java

```diff
@@ -0,0 +1,223 @@
+import org.eclipse.ltk.core.refactoring.participants.DeleteArguments;
+import org.eclipse.ltk.core.refactoring.participants.DeleteParticipant;
+
+/**
+ * @author Simon Jiang
+ */
+public class LiferayGradleModuleProjectDeleteParticipant extends DeleteParticipant {
+
+	public LiferayGradleModuleProjectDeleteParticipant() {
+	}
+
+	@Override
+	public RefactoringStatus checkConditions(IProgressMonitor monitor, CheckConditionsContext context)
+		throws OperationCanceledException {
+
+		return new RefactoringStatus();
+	}
+
+	@Override
+	public Change createChange(IProgressMonitor monitor) throws CoreException, OperationCanceledException {
+		return new RmoveModulePostChange(_deleteProject);
+	}
+
+	@Override
+	public Change createPreChange(IProgressMonitor monitor) throws CoreException, OperationCanceledException {
+		return new RmoveModulePreChange(_deleteProject);
+	}
+
+	@Override
+	public String getName() {
+		return null;
+	}
+
+	public abstract class RemoveModuleChange extends Change {
+
+		public RemoveModuleChange(IProject project) {
+			deleteProject = project;
+		}
+
+		@Override
+		public Object getModifiedElement() {
+			return deleteProject;
+		}
+
+		@Override
+		public String getName() {
+			return "Remove module from workspace project watch list '" + deleteProject.getName() + "'";
+		}
+
+		@Override
+		public void initializeValidationData(IProgressMonitor monitor) {
+		}
+
+		@Override
+		public RefactoringStatus isValid(IProgressMonitor monitor) throws CoreException, OperationCanceledException {
+			return new RefactoringStatus();
+		}
+
+		public abstract Change perform(IProgressMonitor monitor) throws CoreException;
+
+		protected IProject deleteProject;
+
+	}
+
+	public class RmoveModulePostChange extends RemoveModuleChange {
+
+		public RmoveModulePostChange(IProject project) {
+			super(project);
+		}
+
+		@Override
+		public Change perform(IProgressMonitor monitor) throws CoreException {
+			if (_liferayWorkspaceProject == null) {
+				return null;
+			}
+
+			if (ListUtil.isNotEmpty(_projectsToWatch)) {
+				_liferayWorkspaceProject.watch(_projectsToWatch);
+			}
+
+			return null;
+		}
+
+	}
+
+	public class RmoveModulePreChange extends RemoveModuleChange {
+
+		public RmoveModulePreChange(IProject project) {
+			super(project);
+		}
+
+		@Override
+		public Change perform(IProgressMonitor monitor) throws CoreException {
+			if (_liferayWorkspaceProject == null) {
+				return null;
+			}
+
+			IProject workspaceProject = _liferayWorkspaceProject.getProject();
+
+			if (!_projectsToWatch.contains(workspaceProject)) {
+				Stream<IProject> watchStream = _projectsToWatch.stream();
+
+				_projectsToWatch = watchStream.filter(
+					project -> !project.equals(deleteProject)
+				).collect(
+					Collectors.toSet()
+				);
+			}
+			else if (deleteProject.equals(workspaceProject)) {
+				_projectsToWatch.remove(deleteProject);
+			}
+
+			String jobName = workspaceProject.getName() + ":watch";
```

Comment:
Don't hardcode the `":watch"` since that job name may change.  You need to check it using a constant of the same way we do in other places.